src/libraries/Common/src/System/Security/Cryptography/MLDsa.cs

```diff
@@ -1100,14 +1035,12 @@ public static MLDsa ImportMLDsaPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<
             ThrowIfNotSupported();
             ArgumentNullException.ThrowIfNull(algorithm);
 
-            ParameterSetInfo info = ParameterSetInfo.GetParameterSetInfo(algorithm);
-
-            if (source.Length != info.PublicKeySizeInBytes)
+            if (source.Length != algorithm.PublicKeySizeInBytes)
             {
                 throw new CryptographicException(SR.Cryptography_KeyWrongSizeForAlgorithm);
```

Comment:
For `MLKem` we made these `ArgumentException`s. We should consider doing that here too. There is also an existing string resource for the message.